template/apps/api/src/resources/subscription/actions/upgrade.ts

```diff
@@ -0,0 +1,53 @@
+import { z } from 'zod';
+import { stripeService } from 'services';
+
+import { validateMiddleware } from 'middlewares';
+import { AppKoaContext, AppRouter } from 'types';
+
+const schema = z.object({
+  priceId: z.string()
+    .min(1, 'Price id is required')
+    .startsWith('price_', 'Incorrect price id'),
+});
+
+type ValidatedData = z.infer<typeof schema>;
+
+async function handler(ctx: AppKoaContext<ValidatedData>) {
+  const { user } = ctx.state;
+  const { priceId } = ctx.validatedData;
+
+  if (!user.subscription) {
+    ctx.status = 400;
+    ctx.message = 'Subscription does not exist';
+
+    return;
+  }
```

Comment:
You can add this into validator middleware
```suggestion
  if (!user.subscription) {
    ctx.status = 400;
    ctx.message = 'Subscription does not exist';

    return;
  }
```